apps/vmq_server/src/vmq_ranch_config.erl

```diff
@@ -389,7 +389,8 @@ protocol_opts(cowboy_clear, _, Opts) ->
     CowboyRoutes = [{'_', Routes}],
     Dispatch = cowboy_router:compile(CowboyRoutes),
     MaxRequestLength = proplists:get_value(max_request_line_length, Opts, 8000),
-    #{env => #{dispatch => Dispatch}, max_request_line_length => MaxRequestLength, opts => Opts};
+    MaxHeaderValueLength = proplists:get_value(max_header_value_length, Opts, 8000),
```

Comment:
I think you want to add this to the mqttws clause (see L357), not here. I guess the `max_request_line_length` was originally added for HTTP listeners (which get HTTP Rest requests that might need that config).
Try adding both settings to the mqttws clause, and make sure your cookie will be accepted without applying the dynamic config I gave you before. Thanks for testing!

Reply:
added to mqttws but default values (8000) are always extracted here:
```
    MaxRequestLength = proplists:get_value(max_request_line_length, Opts, 8000),
    MaxHeaderValueLength = proplists:get_value(max_header_value_length, Opts, 8000),
``` 
If I set default value high enough, connection to the broker via http upgrade is possible without any further tinkering. As I understand vmq_schema.erl should be changed so that settings could be extracted correctly, but it's a bit to confusing what is considered ws and what is considered http and where, so I am stuck there.